drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -0,0 +1,965 @@
+		if (ret != 0) {
+			LOG_ERR("buffer %p -> out_queue %p err %d", buffer,
+				&stream->out_queue, ret);
+		}
+		if (stream->state == I2S_STATE_RUNNING) {
+			/* allocate new buffer for next audio frame */
+			ret = k_mem_slab_alloc(stream->cfg.mem_slab, &buffer, K_NO_WAIT);
+			if (ret != 0) {
+				if (k_msgq_num_used_get(&stream->in_queue) > 0) {
+					LOG_DBG("wait for data buffer in queue to fill");
+					break;
+				}
+				LOG_ERR("buffer alloc from slab %p err %d",
+					stream->cfg.mem_slab, ret);
+				i2s_mcux_rx_stream_disable(dev, false);
+				stream->state = I2S_STATE_ERROR;
+			} else {
+				const struct i2s_mcux_config *cfg = dev->config;
+				I2S_Type *base = cfg->base;
+
+				dma_reload(stream->dev_dma, stream->channel,
+					   (uint32_t)&base->FIFORD, (uint32_t)buffer,
+					   stream->cfg.block_size);
+				dma_start(stream->dev_dma, stream->channel);
+
+				/* put buffer in input queue */
+				ret = k_msgq_put(&stream->in_queue, &buffer, K_NO_WAIT);
+				if (ret != 0) {
+					LOG_ERR("buffer %p -> in_queue %p err %d",
+					buffer, &stream->in_queue, ret);
+				}
+			}
+		} else {
+			/* Received a STOP/DRAIN trigger */
+			i2s_mcux_rx_stream_disable(dev, true);
+			stream->state = I2S_STATE_READY;
+		}
+		break;
+	case I2S_STATE_ERROR:
+		i2s_mcux_rx_stream_disable(dev, true);
+		break;
+	}
+}
+
+static int i2s_mcux_tx_stream_start(const struct device *dev)
+{
+	int ret = 0;
+	void *buffer;
+	const struct i2s_mcux_config *cfg = dev->config;
+	struct i2s_mcux_data *dev_data = dev->data;
+	struct stream *stream = &dev_data->tx;
+	I2S_Type *base = cfg->base;
+
+	/* retrieve buffer from input queue */
+	ret = k_msgq_get(&stream->in_queue, &buffer, K_NO_WAIT);
+	if (ret != 0) {
+		LOG_ERR("No buffer in input queue to start transmission");
+		return ret;
+	}
+
+	i2s_mcux_config_dma_blocks(dev, I2S_DIR_TX, (uint32_t *)buffer);
+
+	LOG_DBG("dma_slot is %d", stream->dma_cfg.dma_slot);
+	LOG_DBG("channel_direction is %d", stream->dma_cfg.channel_direction);
+	LOG_DBG("complete_callback_en is %d",
+		stream->dma_cfg.complete_callback_en);
+	LOG_DBG("error_callback_en is %d", stream->dma_cfg.error_callback_en);
+	LOG_DBG("source_handshake is %d", stream->dma_cfg.source_handshake);
+	LOG_DBG("dest_handshake is %d", stream->dma_cfg.dest_handshake);
+	LOG_DBG("channel_priority is %d", stream->dma_cfg.channel_priority);
+	LOG_DBG("source_chaining_en is %d", stream->dma_cfg.source_chaining_en);
+	LOG_DBG("dest_chaining_en is %d", stream->dma_cfg.dest_chaining_en);
+	LOG_DBG("linked_channel is %d", stream->dma_cfg.linked_channel);
+	LOG_DBG("source_data_size is %d", stream->dma_cfg.source_data_size);
+	LOG_DBG("dest_data_size is %d", stream->dma_cfg.dest_data_size);
+	LOG_DBG("source_burst_length is %d", stream->dma_cfg.source_burst_length);
+	LOG_DBG("dest_burst_length is %d", stream->dma_cfg.dest_burst_length);
+	LOG_DBG("block_count is %d", stream->dma_cfg.block_count);
+
+	dma_config(stream->dev_dma, stream->channel, &stream->dma_cfg);
```

Comment:
It may be cleaner to put the call to `dma_config` inside `i2s_mcux_config_dma_blocks`. Looking at the function parameters of the two calls

```
i2s_mcux_config_dma_blocks(dev, I2S_DIR_TX, (uint32_t *)buffer);
dma_config(stream->dev_dma, stream->channel, &stream->dma_cfg);
```

It's not clear that one functions prepares data for the other. Also, we would not need to duplicate the numerous `LOG_DBG` entries.

Reply:
Done.